### PR TITLE
Enable user extensions via import path configuration

### LINKED ISSUE #52: Use pythonic configuration for optimizers to allow for user extensions
Currently, the user can specify the objective and initialization methods by simply providing a Python module and object name. However, the optimizer and sampling strategies are [hardcoded](https://github.com/iraikov/dmosopt/blob/main/dmosopt/MOASMO.py#L287).

I wonder if it would make sense to change the semantics of the `optimizer` and `sampling_method` setting to refer to a Python object instead. For example, `"optimizer": "nsga2"` would become `"optimizer": "dmosopt.NSGA2:NSGA2"` etc. In this example, I've used the [entry point](https://setuptools.pypa.io/en/latest/pkg_resources.html#entry-points) notation, but for consistency, it might make more to add a `optimizer_module` option instead.

This would allow users to extend and develop custom strategies by simply pointing to the module. To maintain backward compatibility we could simply keep the current behavior but fall back on importing if the user provides an 'unknown' option.

### PR DESCRIPTION
**Pull request by frthjf**
Allows specification via import path `{option}_name` for:
- [x] optimizer
- [x] surrogate
- [x] sensitivity
- [x] sampling

Renames some options for consistency.

Closes #52.

### CODE REVIEW

**Comment by frthjf** [comment]
Ah, good point. Yes, that's the issue. If I clear out the file it works.

File: pyproject.toml [hunk 1]
```diff
@@ -20,18 +20,18 @@ black = "^22.8.0"
...
-salib = "^1.4.6.1"
+# [tool.poetry.group.extras.dependencies]
+# gpytorch = "^1.9.0"
+# salib = "^1.4.6.1"
 
 [build-system]
 requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
 
-[tool.poetry.extras]
```
Comment (iraikov) [inline pyproject.toml:31]: We should leave the gpflow and gpytorch extra dependencies in until we are ready to have separate packages with support for gpflow and gpytorch.

**Review by iraikov — APPROVED** [review]
I think this is good to go, thanks! I added some dictionaries to resolve default methods in config.py.

**Comment by frthjf** [comment]
Good idea, thanks!

**Pull request merged**